### PR TITLE
allow 'py3-cmd frameclick' on frame modules, fixes #1349

### LINKED ISSUE #1349: frame doesn't respond to py3-cmd click event
In my py3status setup I have the pomodoro module in a frame because I don't want to display it when I am not working. Aiming for a mouseless setup, I am trying to toggle the frame visibility with 
`py3-cmd click 1 'frame pomo'` without any success. Am I doing something wrong ?

Here is the relevant i3status snippet. And of course, it works if I click on the small +/- button.

```
frame pomo {
    format = '{output}{button}'
    pomodoro {
        format = "{mmss}"
    }
}
```

### PR DESCRIPTION
**Pull request by ultrabug**
This fixes issue #1349 for @mrpewpew as well as fixing py-cmd capability to simulate a casual click on a frame ;)

Added documentation as well of course

### CODE REVIEW

File: doc/py3-cmd.rst [hunk 1]
```diff
@@ -56,6 +56,9 @@ You can also specify the button using one of the named shortcuts
     # send a click event to the backlight module with button 5
     py3-cmd scrolldown backlight
 
+    # simulate a click on a frame button to toggle a framed module display
+    py3-cmd click frameclick 'frame pomodoro'
```
Comment (lasers) [inline doc/py3-cmd.rst]: py3-cmd ~~click~~ frameclick 'frame pomodoro' (?)

File: py3status/command.py [hunk 2]
```diff
@@ -83,16 +86,33 @@ def click(self, data):
...
                 instance = module['module'].module_inst
+
+                # this is a frame click request
+                if name == 'frame':
+                    # if the 'frameclick' named click event was requested
+                    # we simulate a click on the frame button instead
+                    if button == 0:
+                        button = 1
+                        instance = '{} button'.format(
+                            ''.join(module_name.split(' ')[1:]))
```
Comment (lasers) [inline py3status/command.py]: Something to mention. Users won't always add `instance`, eg `order += "frame"` would give us `instance = ' button'` (with a leading space). Still work tho. :zap:

**Comment by tobes** [comment]
hey @ultrabug I don't like that this is special for frames.

I've mentioned before but this is what I would really like to see.  Rather than have all these click buttons, and then also the `py3-cmd` triggering clicks.  I'd like modules to define functions, by this I'm not talking about methods on the class, but rather functions that the module provides to the user eg `volume_control` would have `volume_up` `toggle_mute` etc and then we would configure the modules like
```
module {
    format = 'groovy format'
    button 1 = volume_up
    button 2 = toggle_mute
}
```
and `py3-cmd` could learn `py3-cmd action volume_up volume_control`

This way we can have richer controls for both mouse events and hot keys etc

It'd be a bit of work, but I think not too much (updating all modules might take some time).  I've done half the work on the config re-writer that we spoke about, the hard work is done and I'm now happy with the parsing and rewritting model but need to implement the actual config changes.

So I think that we could do this if we agree as a plan

We should do the updated 'fake' event data like you have here though.

**Comment by lasers** [comment]
Updated: https://github.com/lasers/py3status/tree/py3-click

**Comment by ultrabug** [comment]
oh @lasers that branch looks big, care making it an official PR? I'll test it

**Comment by ultrabug** [comment]
#1387 !

**Pull request closed without merging**